product_sequence/models/product_category.py

```diff
@@ -38,22 +38,53 @@ def _prepare_ir_sequence(self, prefix):
         }
         return vals
 
+    def _get_or_create_sequence(self, prefix):
+        seq_vals = self._prepare_ir_sequence(prefix)
+        sequence = self.env["ir.sequence"].search(
+            [("code", "=", seq_vals.get("code")), ("prefix", "=", prefix)]
+        )
+        if not sequence:
+            sequence = self.env["ir.sequence"].create(seq_vals)
+        return sequence
+
     def write(self, vals):
-        prefix = vals.get("code_prefix", False)
-        if prefix:
-            for rec in self:
-                if rec.sequence_id:
-                    rec.sudo().sequence_id.prefix = prefix
-                else:
-                    seq_vals = self._prepare_ir_sequence(prefix)
-                    rec.sequence_id = self.env["ir.sequence"].create(seq_vals)
-        return super().write(vals)
+        sequence_ids_to_check = set()
+        if "code_prefix" in vals:
```

Comment:
why do not hide "code_prefix" if the sequence is selected? 
and are there so many categories to create sequence instead of the user?

Reply:
> why do not hide "code_prefix" if the sequence is selected?

That's how the module is implemented, and since access to ir.sequence requires extra rights IMO it's ok to keep it as it is.

> and are there so many categories to create sequence instead of the user?

Not sure to understand your question, can you please rephrase?

Reply:
> Not sure to understand your question, can you please rephrase?

I meant that automatization is not needed here, that if the user will need to create a sequence, he can check it manually
but as
> access to ir.sequence requires extra rights

understandable